### PR TITLE
usk

### PR DESCRIPTION
**Pull request by sunnya97**

### CODE REVIEW

File: packages/stores/types/ibc-history/polling-status-subscription.d.ts [hunk 1]
```diff
@@ -2,11 +2,11 @@ import { AxiosInstance, AxiosRequestConfig } from "axios";
 /** Polls a `/status` endpoint on a given Axios RPC config and publishes to an arbitrary set of subsribers. */
 export declare class PollingStatusSubscription {
     protected readonly rpc: string;
-    protected readonly rpcConfig?: AxiosRequestConfig<any> | undefined;
+    protected readonly rpcConfig?: AxiosRequestConfig | undefined;
```
Comment (sunnya97) [inline packages/stores/types/ibc-history/polling-status-subscription.d.ts:5]:
```suggestion
    protected readonly rpcConfig?: AxiosRequestConfig<any> | undefined;
```

File: packages/stores/types/ibc-history/polling-status-subscription.d.ts [hunk 2]
```diff
@@ -2,11 +2,11 @@ import { AxiosInstance, AxiosRequestConfig } from "axios";
...
     protected readonly rpcInstance: AxiosInstance;
     protected _subscriptionCount: number;
     protected _handlers: ((data: any) => void)[];
-    constructor(rpc: string, rpcConfig?: AxiosRequestConfig<any> | undefined);
+    constructor(rpc: string, rpcConfig?: AxiosRequestConfig | undefined);
```
Comment (sunnya97) [inline packages/stores/types/ibc-history/polling-status-subscription.d.ts:9]:
```suggestion
    constructor(rpc: string, rpcConfig?: AxiosRequestConfig<any> | undefined);
```

File: packages/web/config/ibc-assets.ts [hunk 3]
```diff
@@ -935,6 +939,21 @@ export const IBCAssetInfos: (IBCAsset & {
...
+    sourceChannelId: "channel-259",
+    destChannelId: "channel-3",
+    coinMinimalDenom: "factory:kujira1qk00h5atutpsv900x202pxx42npjr9thg58dnqpa72f2p7m2luase444a7:uusk",
+    counterpartyMinimalDenom: "factory/kujira1qk00h5atutpsv900x202pxx42npjr9thg58dnqpa72f2p7m2luase444a7/uusk",
+    // depositUrlOverride:
+    //   "https://blue.kujira.app/ibc?destination=osmosis-1&source=kaiyo-1&denom=factory%2Fkujira1qk00h5atutpsv900x202pxx42npjr9thg58dnqpa72f2p7m2luase444a7%2Fuusk",
+    // withdrawUrlOverride:
+    //   "https://blue.kujira.app/ibc?destination=kaiyo-1&source=osmosis-1&denom=ibc%2F44492EAB24B72E3FB59B9FA619A22337FB74F95D8808FE6BC78CC0E6C18DC2EC",
+    // ibcTransferPathDenom:
+    //   "factory:kujira1qk00h5atutpsv900x202pxx42npjr9thg58dnqpa72f2p7m2luase444a7:uusk",
```
Comment (sunnya97) [inline packages/web/config/ibc-assets.ts:953]:
```suggestion
    depositUrlOverride:
      "https://blue.kujira.app/ibc?destination=osmosis-1&source=kaiyo-1&denom=factory%2Fkujira1qk00h5atutpsv900x202pxx42npjr9thg58dnqpa72f2p7m2luase444a7%2Fuusk",
    withdrawUrlOverride:
      "https://blue.kujira.app/ibc?destination=kaiyo-1&source=osmosis-1&denom=ibc%2F44492EAB24B72E3FB59B9FA619A22337FB74F95D8808FE6BC78CC0E6C18DC2EC",
```

File: packages/web/config/ibc-assets.ts [hunk 4]
```diff
@@ -935,6 +939,21 @@ export const IBCAssetInfos: (IBCAsset & {
...
+    
+    isVerified: true,
+  },
```
Comment (sunnya97) [inline packages/web/config/ibc-assets.ts:956]:
```suggestion
  },
```

File: packages/web/stores/assets/index.ts [hunk 5]
```diff
@@ -116,6 +116,11 @@ export class ObservableAssets {
         sourceDenom = ibcAsset.coinMinimalDenom;
       }
 
+      if (ibcAsset.counterpartyMinimalDenom) {
+        sourceDenom = ibcAsset.counterpartyMinimalDenom
+        console.log(sourceDenom)
+      }
+
```
Comment (sunnya97) [inline packages/web/stores/assets/index.ts:123]:
```suggestion
```

File: packages/web/stores/assets/types.ts [hunk 6]
```diff
@@ -19,6 +19,9 @@ export interface IBCAsset {
   // isn't H(source_denom), but rather H(ibc_path)
   ibcTransferPathDenom?: string;
 
+  // If this token has a different minimum denom on the source chain, use this
+  counterpartyMinimalDenom?: string;
+
```
Comment (sunnya97) [inline packages/web/stores/assets/types.ts:24]:
```suggestion
```

**Comment by jonator** [comment]
@sunnya97 is this a stablecoin? If so, could we add the peg mechanism?

**Comment by sunnya97** [comment]
@jonator how do we do that?

**Comment by jonator** [comment]
@sunnya97 

https://github.com/osmosis-labs/osmosis-frontend/blob/75ec3cb5f3af3cd779257f4f64c30594c0d7f8b4/packages/web/config/chain-infos.ts#L2127

**Pull request merged**